### PR TITLE
Kernel error notification

### PR DESCRIPTION
**Pull request by jdetle**
Changed:
- New Kernel Observable does not complete unless there is an error
- Tests finish after both actions are pushed
- Emit an error that is caught by the notification system when spawning fails.

Issue:
When Spawnteract's child process spawning of a kernel fails, it fails asynchronously. `Child_process.spawn` in spawnteract emits an error, on this error we should emit an error action that is caught by the middleware. To test, change the path of a kernel in your kernelspec's kernel.json folder. This should help expose issues for new users.

### CODE REVIEW

File: src/notebook/epics/kernel-launch.js [hunk 1]
```diff
@@ -108,7 +108,11 @@ export function newKernelObservable(kernelSpec: KernelInfo, cwd: string) {
           kernelSpecName,
           kernelSpec,
         });
-        observer.complete();
+
+        spawn.on('error', (error) => {
+          observer.error({ type: 'ERROR', payload: error, err: true });
```
Comment (rgbkrk) [inline src/notebook/epics/kernel-launch.js:113]: Excellent!

File: src/notebook/epics/kernel-launch.js [hunk 2]
```diff
@@ -108,7 +108,11 @@ export function newKernelObservable(kernelSpec: KernelInfo, cwd: string) {
...
+          observer.complete();
```
Comment (rgbkrk) [inline src/notebook/epics/kernel-launch.js:114]: This makes me think we need to complete when the spawn also exits gracefully. Is there another event we should add a callback for?

Comment (rgbkrk) [reply src/notebook/epics/kernel-launch.js:114]: Looks like both `disconnect` and `exit` should be handled - https://nodejs.org/api/child_process.html#child_process_event_exit

**Comment by jdetle** [comment]
Closes #897

File: src/notebook/epics/kernel-launch.js [hunk 3]
```diff
@@ -108,7 +108,17 @@ export function newKernelObservable(kernelSpec: KernelInfo, cwd: string) {
...
+        });
+        spawn.on('exit', () => {
+          observer.complete();
+        });
+        spawn.on('exit', () => {
```
Comment (rgbkrk) [inline src/notebook/epics/kernel-launch.js]: I assume you meant `'disconnect'` here.

**Review by rgbkrk — CHANGES_REQUESTED** [review]

File: src/notebook/epics/kernel-launch.js (hunk 3 above)
Comment (jdetle) [reply src/notebook/epics/kernel-launch.js]: Ugh, yes. Too excited about my first PR in a while...

**Comment by rgbkrk** [comment]
One added note, these callbacks receive more information that we can pump through if we like to inform users (and ourselves better). That can come later, just a note if you want to make this ✨ .

**Comment by jdetle** [comment]
Fixed and rebased typo.

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
This is certainly an improvement on what we had before. Happy to get this in before the next release and kick it around.

**Comment by rgbkrk** [comment]
Code coverage will go down just a bit here, which I'm ok with for the purposes of this PR. Once this passes, we'll merge it. I'll be trying out the release lifecycle next for the main app, making sure our packaging is still ok.

**Comment by jdetle** [comment]
Thank you thank you!

**Pull request merged**